src/main/java/com/microsoft/dhalion/sensors/MultiSensor.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ *
+ * This program is made available under the terms of the MIT License.
+ * See the LICENSE file in the project root for more information.
+ */
+
+package com.microsoft.dhalion.sensors;
+
+import com.microsoft.dhalion.api.ISensor;
+import com.microsoft.dhalion.api.MetricsProvider;
+import com.microsoft.dhalion.conf.Config;
+import com.microsoft.dhalion.conf.Key;
+import com.microsoft.dhalion.core.Measurement;
+import com.microsoft.dhalion.policy.PoliciesExecutor.ExecutionContext;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * The Multi Sensor consumes metrics from the metrics provider and returns them directly in their current form.
+ * It can be used when the metrics returned by the metrics provider do not require any further manipulation before
+ * being consumed by the detectors. Every time the Multi Sensor is invoked it returns metrics that correspond to the
+ * time window between the timestamp of the previous policy invocation and the current timestamp.
+ * <p>
+ * The Multi Sensor takes as input a collection of metric names, the metrics provider, and a set of components.
+ * It collects the values of the given metrics for these components.
+ */
+public class MultiSensor implements ISensor {
```

Comment:
Could you please add unit test cases for this sensor.

Reply:
@ashvina  does it make sense to add a new constructor in BasicSensor only rather than adding a new class for multisensor?